src/output-json-alert.c

```diff
@@ -243,6 +243,10 @@ void AlertJsonHeader(const Packet *p, const PacketAlert *pa, json_t *js)
         }
     }
 
+    if (p->flow) {
+        json_object_set_new(js, "app_proto",
```

Comment:
i guess a flow switch could work, defaulting to 'enabled'. I think the eve block is getting a bit cluttered though, which is why I added 'applayer' to replace the individual protocol switches.

Perhaps we can enable all these things by default and only mention the options in the yaml documentation. That leaves a cleaner yaml.